kong/global.lua

```diff
@@ -201,6 +201,32 @@ function _GLOBAL.init_worker_events()
     return nil, err
   end
 
+  local PAYLOAD_MAX_LEN, PAYLOAD_TOO_BIG_ERR = 65535, "payload too big"
+  local TRUNCATED_PREFIX = ", truncated payload: "
+  local DEFAULT_TRUNCATED_PAYLOAD = TRUNCATED_PREFIX .. "not a serialized object"
+  local LEN = (#PAYLOAD_TOO_BIG_ERR + #DEFAULT_TRUNCATED_PAYLOAD) * 2
+
+  -- There is a limit for the payload size that events lib allows to send,
+  -- we overwrite `post` method to truncate the payload and send it again
+  -- when we get error message: "payload too big"
+  local native_post = worker_events.post
+  worker_events.post = function (source, event, data, unique)
+    local ok, err = native_post(source, event, data, unique)
+    -- exceeds the upper limit for the size of the payload
+    if err == PAYLOAD_TOO_BIG_ERR then
+      if type(data) == "string" then
+        -- truncate the payload and send it again  
+        data = PAYLOAD_TOO_BIG_ERR .. ", truncated payload: " .. string.sub(data, 1, PAYLOAD_MAX_LEN - LEN)
+      else
+        data = PAYLOAD_TOO_BIG_ERR .. DEFAULT_TRUNCATED_PAYLOAD
```

Comment:
Should we send this error message as event data? should we tell the user the call fails?

Reply:
In some cases, we just need the subscriber to be informed by particular events without caring about the specific payload of the events, for instance, some caches of entity rely on DAO:crud event to refresh cache.